### ISSUE TITLE
LogRecord modified

### ISSUE THREAD

**Issue by mohammedtabish0** [rahulrai-in/otel-logs-demo#1]
If I were to modify the log record in one exporter then the same changes are getting reflected to other exporters which are after that in the pipeline. 

for e.g If my pipeline looks like AppInsightsExporter -> ConsoleExporter
I were to mask variables in AppInsightsExporterand I publish it then the same LogRecord is passed down with the masked variables to the ConsoleExporter.

**Comment by rahulrai-in** [comment]
The exporters are not related to the processors. If you look at the diagram in the [article](https://www.twilio.com/en-us/blog/build-a-logs-pipeline-in-dotnet-with-opentelemetry), the processors are executed before the telemetry is sent to the exporter.
If you want to create two pipelines to customize the behavior of the processors, I'd recommend using OpenTelemetry collector and creating 2 pipelines through configuration. Here is how it is done: https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/design.md

**Comment by mohammedtabish0** [comment]
There was a mistake,It is not exporter which I am taling about but rather processor.In my .NET application I am using a custom appinsights processor and a custom console processor where I am overriding the `OnEnd()` method modifying the `LogRecord` object and then calling the `base.OnEnd(data)` at the end.I have registered them in the previous stated order and the modified logrecord instance in custom appsights processor is getting passed to custom console processor. 

-It's a production code in which I am facing this issue otherwise I would have pasted. If you would like an example I can create and share (facing other issues as well like there is no extension method to add exporter of appsinghts library like `AddConsoleExporter()` so I had to use reflection to get exporter instance and pass it down in processor's instance and I have extend simpleLogExporterProcessor instead of the Batch one which is the recommended way.)

**Comment by rahulrai-in** [comment]
If you can't use the Otel collector, can you try creating 2 Otel log providers using the function: `builder.Logging.AddOpenTelemetry(logging => ...)`. Configure one with the processor and one without the processor based on your need.

**Comment by mohammedtabish0** [comment]
If I understood correctly what you are saying is I call this `builder.Logging.AddOpenTelemetry(logging => ...)` two times and then in one call, I configure the AppInsights processor and in the other one I register console processor only and this will make sure that the LogRecord object modified in appinsights would be completely seperate from the console processor's LogRecord object i.e this will create two different "pipelines" and If I use DI to inject `ILogger` instance it would work?

**Comment by rahulrai-in** [comment]
Yes. I expect that it will add 2 logging providers with their own behaviours.
